bolt-design-system/bolt
# Bolt Release v2.2.0

## Enhancements
- Added align prop for `bolt-share`. #922 
- Significantly refactored and updated Bolt's build tools (900% speed increase!) #923
- Maintenance updates and improvements to our Bolt-integrated Drupal Lab instance #894
- Added brand new Bolt version selector dropdown to the Docs site header + initial implementation of the new `<bolt-select>` custom element. #906
- Update Lerna to latest version #932
- Break out the mutation observer logic currently used in the button component into it's own standalone JS utility library for reuse in Bolt core. #935
- Backported upstream Pattern Lab UIKit improvements #911
- Add pattern lab demos for grid component row_gutter and vinset #919
- Updated the new Bolt version selector on the docs site to optionally allow the Bolt version being selected to open in a new tab if the ⌘ (CMD) / meta key is being pressed down at the same time. #931
- Added additional logging to the Button component's 3rd party JS integration demo in Pattern Lab. #933
- Update the default sourcemap config settings used in Bolt's Webpack's devtool configuration for local dev #938
- Adds a new opacity scale setting to standardize opacities throughout the design system #944
- Update `@bolt/build-tools` to skip Bolt version check during non-prod builds #949
- Updated the Bolt Getting Started Guide to help improve the initial developer onboarding process #954
- `@bolt/core updates` to help address cross-browser rendering inconsistencies by switching over from HyperHTML to Google's Lit-HTML library + updating existing components accordingly #951 
- Button component internal logic simplified to help improve cross-browser rendering + help reduce the number of moving parts #937
- Add cross browser polyfills for Template and Symbol to @bolt/core/polyfills (required for lit-html) #951
* Added two new React-inspired Javascript additions to `@bolt/core`: `withContext` and `defineContext`.
    - This enhancement allows parent components to define certain props that child components can now easily grab the value for + get alerted when the parent's data changes.

> While not yet used by any components at the moment, these updates support several upcoming component enhancements (such as the in-progress work on the Card, Link, and List components) that involve contextual data and internal state management.
> 
> Check out this [recent CSS Tricks article ](https://css-tricks.com/digging-into-react-context/) to learn more about the ideas behind this!

## Fixes
- Fixed `bolt-share`'s tight spacing issue in FF. #922 
- Fixed a typo in the bolt-list twig file about the inset prop. #929
- Make Webpack build errors more descriptive regardless of reporting level #936 
- Fix the dynamic import path to the "native shim" polyfill due to upstream changes in the main webcomponentsjs repo #934 
- Fix form label theming, most notably when form labels appear in cards in IE 11 #930
- Add a variant of button for use in forms that does not invoke any Bolt javascript #940 
- Add support for the grid component's row_gutter parameter in IE 11. #919 
- Fix an issue with certain icons not rendering in Safari browsers #942 
- Add a cache buster query string to the Bolt docs site head and footer’s main CSS and JS bundles #941 
- Prevent bolt-nav-priority from rendering when no links are provided (prevents the chevron icon in navbar from appearing in Safari when there are no links to display) #945 
- Addressed a Pattern Lab-specific issue with the async loaded Pattern Lab CSS (via LoadCSS) that had been causing an infinite loading loop in IE 11 -- interfering in reliably testing out JS updates #951

## Deprecations
Three opacity-related utility classes are deprecated and will be removed in the next major Bolt release, `v3.0.0`.  

Replace them with the recommended classes shown in the table below or one of the other classes documented at https://feature-opacity-scale.boltdesignsystem.com/pattern-lab/?p=styleguide-opacity

| Deprecated utility classes | Suggested replacement
| --------------------- | -----
| `.u-bolt-opacity-75` |`.u-bolt-opacity-80`
| `.u-bolt-opacity-50` |`.u-bolt-opacity-60`
| `.u-bolt-opacity-25` |`.u-bolt-opacity-20`
